man/mkdocs/docs/python_intro.md

```diff
@@ -114,116 +108,121 @@ color_scheme = """
+| [mapcalc3d](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.raster3d.mapcalc3d) | Execute a 3D raster algebra expression |
+
+Here is an example how to use these helper functions:
 
 ```python
-# Get metadata about the elevation raster
-gs.raster_info('elevation')
-
-# Create a new raster map with the mapcalc expression
-gs.mapcalc(expression="elevation_2x = elevation * 2")
-```
+# Get region resolution to get cell size
+region = gs.region()
+cell_area = region["nsres"] * region["ewres"]
 
-!!! grass-tip "Why not use the tool directly?"
-    The `raster` module provides a more functional approach to working with raster
-    maps in GRASS.
+# Compute volume with raster algebra
+gs.mapcalc(f"volume = elevation * {cell_area}")
 
-<!-- markdownlint-disable-next-line line-length -->
-[Full Documentation :material-arrow-right-bold:](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#module-script.raster){ .md-button }
-
-### Vectors
+# Get number of points in a vector map
+num_points = gs.vector_info_topo("hospitals")["points"])
+```
 
-The [grass.script.vector](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#module-script.vector)
-library provides functional access to vector metadata
-and processing tools. Some of the core functions include:
+### NumPy interface
 
-| Function            | Description                                      |
-|---------------------|--------------------------------------------------|
-| [vector_columns](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.vector.vector_columnsr)    | Get the columns of a vector map                  |
-| [vector_info](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.vector.vector_info)       | Get information about a vector map               |
-| [vector_db](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.vector.vector_db)         | Get the database connection of a vector map      |
-| [vector_db_select](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.vector.vector_db_select)  | Select data from a vector map                    |
-| [vector_info_topo](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.vector.vector_info_topo)  | Get topology information about a vector map      |
+The GRASS Python API includes a NumPy interface that allows you to read
+and write raster data as NumPy arrays.
+This makes it easy to integrate GRASS with the broader Python scientific
+stack for advanced analysis and custom modeling.
+Using *[grass.script.array](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.array.array)*
+and *[grass.script.array3d](https://grass.osgeo.org/grass-stable/manuals/libpython/script.html#script.array.array3d)*,
+you can load raster maps into memory, perform array-based operations,
+and write results back as a GRASS raster or 3D raster.
```

Comment:
The array actually are not in memory, at least not initially. The intro covers only the "jump to NumPy" case but not the "jump to GRASS" case. Here is a suggestion to fix both:

```suggestion
you can switch between GRASS raster maps and NumPy arrays, run GRASS tools,
and perform array-based operations as needed.
It works for rasters as well as for 3D rasters in GRASS.
```